db_API.py

```diff
@@ -0,0 +1,236 @@
+
+        cursor = connection.cursor()
+        cursor.execute(sql)
+        connection.commit()
+
+    connection.close()
+
+
+def insert_row(table_name, columns, values):
+    connection = dbconnection.getdbconnection()
+    try:
+        sql = 'INSERT INTO %s (%s) VALUES (' % (table_name, ", ".join(columns))
+
+        for value in values:
+
+            if type(value) == str:
+                sql += '"%s"' % value + ", "
+            else:
+                sql += str(value) + ", "
+
+        sql = sql[:-2] + ")"
+        # print(sql)
+
+        cursor = connection.cursor()
+        cursor.execute(sql)
+        connection.commit()
+
+    except pymysql.err.IntegrityError:
+        pass
+
+    connection.close()
+
+
+def fetch_one(table_name, select_columns, where_dict):
+    connection = dbconnection.getdbconnection()
+
+    sql = 'SELECT %s FROM %s WHERE ' % (", ".join(select_columns), table_name)
+    for key in where_dict.keys():
+        if type(where_dict.get(key)) == str:
+            sql += '%s = "%s" AND ' % (key, where_dict.get(key))
+        # elif type(where_dict.get(key)) == int:
+        #     sql += '%s = "%s" AND ' % (key, where_dict.get(key))
+        else:
+            sql += '%s = %s AND ' % (key, where_dict.get(key))
+
+    sql = sql[:-5]
+    # print(sql)
+    cursor = connection.cursor()
+    cursor.execute(sql)
+
+    row = cursor.fetchone()
+
+    connection.close()
+
+    return row
+
+
+#    Get all genes information and return results as tuple
+def getGene_all():
+    connection = dbconnection.getdbconnection()
+    sql = 'SELECT Accession_No, Gene_Identifier, Chromosomal_Location, Organism, Sequence_Length, DNA_Sequence FROM Gene'
+    cursor = connection.cursor()
+    cursor.execute(sql)
+    for row in cursor.fetchall():
+        yield row
+
+    connection.close()
+    # connection.commit()
+
+
+
+#    Get Gene info by accession number and return results as tuple
+def getGene_acno(acno):
+    connection = dbconnection.getdbconnection()
+    sql = 'SELECT Accession_No, Gene_Identifier, Chromosomal_Location, Organism, Sequence_Length, DNA_Sequence ' \
+          'FROM Gene WHERE Accession_No = "%s"' % acno
+    cursor = connection.cursor()
+    cursor.execute(sql)
+
+    for row in cursor.fetchall():
+        yield row
+
+    # connection.commit()
+    connection.close()
+    
+
+#    Get Gene info by gene idenitifer and return results as tuple
+def getGene_gi(gene_identifier):
+    connection = dbconnection.getdbconnection()
+    sql = 'SELECT Accession_No, Gene_Identifier, Chromosomal_Location, Organism, Sequence_Length, DNA_Sequence ' \
+          'FROM Gene WHERE Gene_Identifier = "%s" ' % gene_identifier
+    cursor = connection.cursor()
+    cursor.execute(sql)
+
+    for row in cursor.fetchall():
+        yield row
+    
+    # connection.commit()
+    connection.close()
+
+
+#    Get all Protein information and return results as tuple
+def getProtein_all():
+    connection = dbconnection.getdbconnection()

```

Comment:
it would be much better for the scoring if you include the inital comment inside the `def` body as per `PEP 257` e.g.
```
def getProtein_all():
    """ Fetches protein information from the database
          args: none
          return: row (tuple) [tuple with what?]
    """
```